### PR TITLE
Benchmark refactor

### PR DESCRIPTION
**Pull request by mullermp**
Refactor benchmarking to use a shared tooling repository

### CODE REVIEW

**Review by alextwoods — APPROVED** [review]
Nice, looks good!

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -17,13 +17,14 @@ jobs:
...
+      GH_EVENT: ${{github.event_name}}
       GH_REF: ${{github.head_ref}}
       GH_REPO: ${{github.repository}}
-      GH_EVENT: ${{github.event_name}}
+      EXECUTION_ENV: github-action
     strategy:
       fail-fast: false
       matrix:
-        ruby: [jruby-9.2, 2.6, 2.7, 3.2, 3.3]
+        ruby: [jruby-9.4, 3.3]
```
Comment (alextwoods) [inline .github/workflows/benchmark.yml]: Probably a bigger discussion on what versions we want to track against.  I see a point in not necessarily tracking this data for every ruby version, esp on PRs.  But for continuity over time, we likely want to keep a complete list for releases (ie, maybe we need separate tasks that run on version-3 branch vs PRs?).  But we can address that in the future.

File: tasks/gems/cloudwatchlogs.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        'Aws::CloudWatchLogs'
+      end
+
+      def operation_benchmarks
+        {
+          put_log_events_small: {
+            setup: proc do |_client|
+              { log_group_name: 'log_group', log_stream_name: 'log_stream', log_events: (0...5).map do |i|
+                                                                                          { timestamp: Time.now.to_i, message: "test log event #{i}" }
+                                                                                        end }
```
Comment (jterapin) [inline tasks/gems/cloudwatchlogs.rb]: Could cut down the line length like follows? For this and others alike in this PR
```
def operation_benchmarks
  {
    put_log_events_small: {
      setup: proc do |_client|
        {
          log_group_name: 'log_group',
          log_stream_name: 'log_stream',
          log_events: (0...5).map do |i|
            { timestamp: Time.now.to_i, message: "test log event #{i}" }
          end
        }
      end
      ...
```

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -32,29 +33,31 @@ jobs:
...
-      - name: configure aws credentials
+      - name: Configure AWS credentials
         uses: aws-actions/configure-aws-credentials@v4
         with:
-          role-to-assume: arn:aws:iam::469596866844:role/aws-sdk-ruby-performance-reporter
-          role-session-name: benchmark-report
-          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::373952703873:role/BenchmarkReporter
+          role-session-name: benchmark-reporter
+          aws-region: us-east-1
```
Comment (jterapin) [inline .github/workflows/benchmark.yml:58]: Curious to know why we are switching regions here?

**Review by jterapin — APPROVED** [review]

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (mullermp) [reply .github/workflows/benchmark.yml]: We can do latest versions or all supported versions.

File: .github/workflows/benchmark.yml (hunk 3 above)
Comment (mullermp) [reply .github/workflows/benchmark.yml:58]: No particular reason other than our other team resources are in us-east-1 here.

File: tasks/gems/cloudwatchlogs.rb (hunk 2 above)
Comment (mullermp) [reply tasks/gems/cloudwatchlogs.rb]: Fixed up all gem files.

**Pull request merged**